staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -99,12 +100,43 @@ type objState struct {
 	stale bool
 }
 
+// ListErrorAggregator aggregates the error(s) that the LIST operation
+// encounters while retrieving object(s) from the storage
+type ListErrorAggregator interface {
+	// Aggregate aggregates the given error from list operation
+	// key: it identifies the given object in the storage.
+	// err: it represents the error the list operation encountered while
+	// retrieving the given object from the storage.
+	// done: true if the aggregation is done and the list operation should
+	// abort, otherwise the list operation will continue
+	Aggregate(key string, err error) bool
+
+	// Err returns the aggregated error
+	Err() error
+}
+
+// DefaultListErrorAggregatorFactory returns the default list error
+// aggregator that maintains backward compatibility, which is abort
+// the list operation as soon as it encounters the first error
+func DefaultListErrorAggregatorFactory() ListErrorAggregator { return &abortOnFirstError{} }
```

Comment:
I still think this is all 2 levels too meta. A factory for a leaking abstraction.

Reply:
we need to switch to corrupt object error aggregation when the feature is enabled https://github.com/kubernetes/kubernetes/pull/127513/commits/d09040330a93994592a100899bf8aca208448f54

The other option is to make these changes inline inside store.go with `if feature enabled {} else {}` which i wanted to avoid. Also when the feature gate is removed we won't need this factory, we will have just one type of aggregator. Thoughts?

Reply:
@sttts were you satisfied on this thread?